vault/resource_pki_secret_backend_issuer.go

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) HashiCorp, Inc.
```

Comment:
This resource does not create issuers, since that is not possible via the Vault's API. Issuers are generated through the [generate root](https://registry.terraform.io/providers/hashicorp/vault/latest/docs/resources/pki_secret_backend_root_cert) and [set-signed](https://registry.terraform.io/providers/hashicorp/vault/latest/docs/resources/pki_secret_backend_intermediate_set_signed) endpoints, and are passed down to resources.

My initial approach was to try clean up issuers in the aforementioned resources themselves, but this leads to some other issues:
- Set signed does not support a delete endpoint
- Users have no ability to tack and update any issuer information
- TF can track the generate root and set signed resource, but has no idea of how to clean up dangling issuers once `terraform destroy` is called. This leads to errors in cleanup of keys as well even though everything has been destroyed, because the key is still "in use" by an existing, dangling issuer. Issuer cleanup via TFVP seems paramount

Due to all the above, this resource takes the approach of expecting that an issuer has already been created. In the create method, we ensure the issuer exists. There are no write calls made to Vault in create. We can then also give the users the ability to update any issuer fields they'd like (currently I only added `issuer_name` but can add in other fields if we decide this approach works). Most importantly, this resource has the ability to track issuer changes, and can delete issuers and clean them up whenever a `destroy` operation is called

Reply:
Nice! thanks

Reply:
Yup a great start for the issuer type, much appreciated!